File: task5.cpp
#include <iostream>
#include <iomanip>
#include <cmath>

using namespace std;

// Функция моделирования остывания кофе
void simulateCooling(double T0, double Ts, double r, double dt, double totalTime) {
    double T = T0;
    int steps = static_cast<int>(totalTime / dt);

    // Заголовок таблицы
    cout << "\n+--------------+-------------------------+\n";
    cout << "| Время (с)    | Температура кофе (°C)   |\n";
    cout << "+--------------+-------------------------+\n";

    cout << fixed << setprecision(2);

    for (int i = 0; i <= steps; ++i) {
        double time = i * dt;

        cout << "| " << right << setw(12) << time << " | " << right << setw(23) << T << " |\n";

        T += -r * (T - Ts) * dt;
    }

    cout << "+--------------+-------------------------+\n";
}

int main() {
    double T0, Ts, r, dt, totalTime;

    // Ввод параметров
    cout << "Введите начальную температуру кофе (T0): ";
    cin >> T0;

    cout << "Введите температуру окружающей среды (Ts): ";
    cin >> Ts;

    cout << "Введите коэффициент остывания (r): ";
    cin >> r;

    cout << "Введите шаг по времени (dt): ";
    cin >> dt;

    cout << "Введите общее время моделирования (в секундах): ";
    cin >> totalTime;

    // Запуск моделирования
    simulateCooling(T0, Ts, r, dt, totalTime);

    return 0;
}

File: task5.py
def simulate_cooling(T0, Ts, r, dt, total_time):
    T = T0
    steps = int(total_time / dt)
    print("\n+--------------+-------------------------+")
    print("| Время (с)    | Температура кофе (°C)   |")
    print("+--------------+-------------------------+")
    for i in range(steps + 1):
        time = i * dt
        print(f"| {time:12.2f} | {T:23.2f} |")
        T += -r * (T - Ts) * dt
    print("+--------------+-------------------------+")

def main():
    T0 = float(input("Введите начальную температуру кофе (T0): "))
    Ts = float(input("Введите температуру окружающей среды (Ts): "))
    r = float(input("Введите коэффициент остывания (r): "))
    dt = float(input("Введите шаг по времени (dt): "))
    total_time = float(input("Введите общее время моделирования (в секундах): "))
    simulate_cooling(T0, Ts, r, dt, total_time)

if __name__ == "__main__":
    main()
